app/uk/gov/hmrc/eusubsidycompliance/controllers/EmailController.scala

```diff
@@ -0,0 +1,77 @@
+package uk.gov.hmrc.eusubsidycompliance.controllers
+
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, AnyContent, ControllerComponents, Result}
+import uk.gov.hmrc.eusubsidycompliance.controllers.actions.Authenticator
+import uk.gov.hmrc.eusubsidycompliance.models.types.EORI
+import uk.gov.hmrc.eusubsidycompliance.models.{ApproveEmailAsVerifiedByEoriRequest, ApproveEmailByVerificationIdRequest, ConnectorError, EmailCache, StartEmailVerificationRequest, VerifiedEmailResponse}
+import uk.gov.hmrc.eusubsidycompliance.persistence.{EoriEmailRepository, InitialEmailCache}
+import uk.gov.hmrc.eusubsidycompliance.util.UuidProvider
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class EmailController @Inject() (
+  cc: ControllerComponents,
+  authenticator: Authenticator,
+  eoriEmailRepository: EoriEmailRepository,
+  uuidProvider: UuidProvider
+)(implicit ec: ExecutionContext)
+    extends BackendController(cc) {
+
+  def startVerification: Action[JsValue] = authenticator.authorisedWithJson(parse.json) { implicit request => _ =>
+    withJsonBody[StartEmailVerificationRequest] { startEmailVerificationRequest: StartEmailVerificationRequest =>
+      val initialEmailCache = InitialEmailCache.apply(
+        eori = startEmailVerificationRequest.eori,
+        verificationId = uuidProvider.getRandom.toString,
+        email = startEmailVerificationRequest.emailAddress,
+        verified = false
+      )
+
+      val eventualMaybeEmailCache = for {
+        _ <- eoriEmailRepository.addData(initialEmailCache)
+        entry <- eoriEmailRepository.get(startEmailVerificationRequest.eori)
+      } yield entry
+
+      remapGetEmailVerificationResponse(eventualMaybeEmailCache, InternalServerError)
+    }
+  }
```

Comment:
The controller needs to match EscService as that was the guardian of mongo which we moved so we only care about updating varified flag.

https://github.com/hmrc/eu-subsidy-compliance-frontend/pull/346/files#diff-43b6d40e642b47212103844ca7bd3af4c18ede6b17fc64cc8b294ab3d137465a